modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/UpgradeJavaAddFDSTableSchemaFieldCheck.java

```diff
@@ -0,0 +1,103 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.source.formatter.check;
+
+import com.liferay.portal.kernel.util.StringUtil;
+import com.liferay.source.formatter.parser.JavaClass;
+import com.liferay.source.formatter.parser.JavaClassParser;
+import com.liferay.source.formatter.parser.JavaMethod;
+import com.liferay.source.formatter.parser.JavaTerm;
+
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * @author Michael Cavalcanti
+ */
+public class UpgradeJavaAddFDSTableSchemaFieldCheck extends BaseFileCheck {
+
+	@Override
+	protected String doProcess(
+			String fileName, String absolutePath, String content)
+		throws Exception {
+
+		if (!fileName.endsWith(".java")) {
+			return content;
+		}
+
+		JavaClass javaClass = JavaClassParser.parseJavaClass(fileName, content);
+
+		List<String> importNames = javaClass.getImportNames();
+
+		if (!importNames.contains(
+				"com.liferay.frontend.data.set.view.table." +
+					"FDSTableSchemaBuilder")) {
+
+			return content;
+		}
+
+		return _replaceAddFDSTableSchemaFieldMethod(content, javaClass);
+	}
+
+	private String _checkMethodCall(String content, String javaMethodContent) {
+		Matcher addFDSTableSchemaFieldMatcher =
+			_addFDSTableSchemaFieldPattern.matcher(javaMethodContent);
+
+		while (addFDSTableSchemaFieldMatcher.find()) {
+			javaMethodContent = StringUtil.replace(
+				javaMethodContent, addFDSTableSchemaFieldMatcher.group(0),
+				_replaceMethodCall(
+					content, addFDSTableSchemaFieldMatcher.group(0)));
+		}
+
+		return javaMethodContent;
+	}
+
+	private String _replaceAddFDSTableSchemaFieldMethod(
+		String content, JavaClass javaClass) {
+
+		for (JavaTerm childJavaTerm : javaClass.getChildJavaTerms()) {
+			if (!childJavaTerm.isJavaMethod()) {
+				continue;
+			}
+
+			JavaMethod javaMethod = (JavaMethod)childJavaTerm;
+
+			String javaMethodContent = javaMethod.getContent();
+
+			content = StringUtil.replace(
+				content, javaMethodContent,
+				_checkMethodCall(javaClass.getContent(), javaMethodContent));
```

Comment:
@michaelmdelcavalcanti, this method name is too generic, how about we rename it to `replaceAddFDSTableSchemaField`, for example, and merge this method and the method `_replaceMethodCall` into a single method?